src/Host/Client/Impl/Definitions/IRBlobData.cs

```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.R.Host.Client {
+    public interface IRBlobData : IRBlob {
+        RBlobKind Kind { get; }
```

Comment:
Is this actually necessary? It feels like it simply duplicates the interface hierarchy - i.e. one can check whether it's a file or not via `... is IRBlobFile`.